### PR TITLE
Keep audit of policy comment changes

### PR DESCRIPTION
**Pull request by allbecauseyoutoldmeso**
### Description of change

- Update `Policy` model so policies have many comments instead of one.
- Update logic of policy class form so that a new comment is created if the assessor edits the comment.
- Update logic of policy class review form so that a new comment is created if the reviewer edits the comment.
- Show previous comments section on both forms.
- Add `deleted_at` to `comments`. When an assessor 'deletes' a comment add the `deleted_at` time rather than actually destroying the record. Once 'deleted' the record is visible in the 'previous comments' section but is not treated as the 'current' comment.

### Story Link

https://trello.com/c/Bo5YCoX4/1378-show-the-assessor-reviewer-any-previous-versions-of-comments-against-legislation-in-a-revealable-details-section

### Screenshots

Assessor form:

Screenshot 2022-12-19 at 11 56 07

Reviewer form:

Screenshot 2022-12-19 at 11 53 52

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -100,7 +100,7 @@ GEM
     bindex (0.8.1)
     bootsnap (1.9.1)
       msgpack (~> 1.0)
-    brakeman (5.1.1)
+    brakeman (5.4.0)
```
Comment (allbecauseyoutoldmeso) [inline Gemfile.lock:103]: Earlier versions can't parse begin-less ranges 😮

File: app/views/policy_classes/_comment.html.erb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+<%= content_tag(:div, data: { controller: "delete-record" }) do %>
```
Comment (allbecauseyoutoldmeso) [inline app/views/policy_classes/_comment.html.erb:1]: Most of this file is just `app/views/policy_classes/_comment_form.html.erb` renamed!

File: spec/components/policy_classes/comment_field_component_spec.rb [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      render_inline(component)
+    end
+
+    it "renders field with submitted text" do
+      expect(page).to have_field("Add comment", with: "test")
+    end
+  end
+
+  context "when an invalid blank value has been submitted" do
+    let!(:comment) do # rubocop:disable RSpec/LetSetup
```
Comment (allbecauseyoutoldmeso) [inline spec/components/policy_classes/comment_field_component_spec.rb:59]: I don't know why rubocop this 'isn't referenced in the test'. It's referenced on line 10 and the test blows up if I replace it with `before do...`

File: app/components/policy_classes/comment_field_component.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+    def invalid?
+      @invalid ||= new_comment.present? && !new_comment.valid?
+    end
+
+    def new_comment
+      @new_comment ||= policy.comments.find(&:new_record?)
+    end
+  end
+end
```
Comment (allbecauseyoutoldmeso) [inline app/components/policy_classes/comment_field_component.rb:33]: Rendering errors for the nested attributes in nested forms 😓

File: app/views/shared/policy_classes/_comment.html.erb [hunk 5]
```diff
@@ -1,5 +1,5 @@
 <% if comment.present? %>
-  <h4 class="govuk-heading-s policy-comment-title">
+  <h4 class="govuk-heading-s govuk-!-margin-bottom-0">
```
[the remark this answers is not in the document]
Comment (notapatch) [reply app/views/shared/policy_classes/_comment.html.erb:2]: With utility classes one of the most important is spacing as when I tried to reuse css (which I rarely succeeded at for more than the most basic components) it varied the most..
Tailwind would have that as `mb-0`.

**Review by notapatch — APPROVED** [review]
I'm not sure how this works, but I don't have any issue with the code itself! Beyond my ignorance of the solution.

File: app/controllers/comments_controller.rb [hunk 6]
```diff
@@ -1,35 +1,39 @@
 # frozen_string_literal: true
 
 class CommentsController < AuthenticationController
-  def update
-    comment.update(comment_params)
-    render(json: { partial: update_comment_form })
+  def create
+    new_comment = policy.comments.new(comment_params)
+    new_comment = policy.comments.new if new_comment.save
```
[the remark this answers is not in the document]
Comment (allbecauseyoutoldmeso) [reply app/controllers/comments_controller.rb:6]: Yes. If it doesn't save, render the form with the comment we've just tried to create, with the associated errors. If it does save, render the form for a shiny new comment. 

It's sort of mimicking the normal Rails behaviour, the difference being that we're staying on the same page!

**Pull request merged**